capsul/process/node.py

```diff
@@ -262,6 +269,9 @@ def add_field(self, name, type_, default=undefined, metadata=None,
         }
         # generate plug with input parameter and identifier name
         self._add_plug(parameter)
+        # notify now the new field/plug
+        if self.enable_notification:
+            self.on_fields_change.fire()
```

Comment:
This answer my previous comment. I did not noticed at first sight it was in the same method (GitHub added a visual separation).

Reply:
yes I had to delay the notification because now that `Process` and `Node` are the same, notification must fire after the `Node` part of `add_field()` (namely, add the new plug) is done, otherwise observers will be notified while plugs are not correct yet.